executor/executor.go

```diff
@@ -5,175 +5,235 @@ package executor
 
 import (
 	"sync"
+	"sync/atomic"
 
 	"github.com/ava-labs/avalanchego/utils/set"
 
 	"github.com/ava-labs/hypersdk/state"
+
+	uatomic "go.uber.org/atomic"
 )
 
-const defaultSetSize = 8
+// maxDependencies must be greater than the maximum number of dependencies
+// any single task could have. This is used to ensure a dependent task
+// does not begin executing a task until all dependencies have been enqueued.
+const maxDependencies = 100_000_000
```

Comment:
I think the last tweak we should make is to make this a param you provide in `New`. I think this const is "jank".